apps/gitness/src/pages/pipeline-list.tsx

```diff
@@ -62,9 +72,26 @@ export default function PipelinesPage() {
     )
   }
 
+  //fetch projects api to get the list of projects
+  const { data: projects, isLoading } = useMembershipSpacesQuery({
+    queryParams: { page: 1, limit: 30, sort: 'identifier', order: 'asc' }
+  })
+
+  //prevent rendering the page until the projects are loaded
+  if (isLoading) {
```

Comment:
@jessieyu777 blocking the entire page from loading while the project list loads could be a bit too dramatic? Maybe the dropdown is where the project loading state should be communicated?

Reply:
@joetaylor-harness That's a temporary status for not breaking the page from loading because of the gap of getting data. The error will show from the component side since if the projects props was empty. so I wrote a condition function to the high level to this page first. Already discussed it in the meeting today. We decided to make this branch merge first, and then figure out what is the best status for showing this loading since we didn't get the design-look for this details.

Reply:
Currently it look like this since I call api separately. 
<img width="1728" alt="Screenshot 2024-09-11 at 2 15 39 PM" src="https://github.com/user-attachments/assets/8756ad60-f581-4dcc-9d52-aa5c24a31190">